sdk/ai/azure-ai-generative/azure/ai/generative/synthetic/qa.py

```diff
@@ -22,27 +21,62 @@
     print("In order to use qa, please install the 'qa_generation' extra of azure-ai-generative")
     raise e
 
+try:
+    import pkg_resources
+    openai_version_str = pkg_resources.get_distribution("openai").version
+    openai_version = pkg_resources.parse_version(openai_version_str)
+    import openai
+    if openai_version >= pkg_resources.parse_version("1.0.0"):
+        _RETRY_ERRORS = (
+            openai.APIConnectionError ,
+            openai.APIError,
+            openai.APIStatusError
+        )
+    else:
+        _RETRY_ERRORS = (
+            openai.error.ServiceUnavailableError,
+            openai.error.APIError,
+            openai.error.RateLimitError,
+            openai.error.APIConnectionError,
+            openai.error.Timeout,
+        )
+        
+except ImportError as e:
+    print("In order to use qa, please install the 'qa_generation' extra of azure-ai-generative")
```

Comment:
Use  a logger or warning for this.

Reply:
I have a follow up work item to improve telemetry from the package, I will add a logger as part of those changes. I will get this in the follow up